transcoder/src/keyframes.go

```diff
@@ -199,12 +212,18 @@ func getVideoKeyframes(path string, video_idx uint32, kf *Keyframe) error {
 			return err
 		}
 
-		// Before, we wanted to only save keyframes with at least 3s betweens
-		// to prevent segments of 0.2s but sometimes, the -f segment muxer discards
-		// the segment time and decide to cut at a random keyframe. Having every keyframe
-		// handled as a segment prevents that.
+		// The -f hls encoder will cut at the next keyframe found every X seconds.
+		// To be sure there is no mismatch between what -f hls outputs and what we think segments are, we need to make
+		// sure every keyframes is at least `OptimalFragmentDuration` away from the next.
+		//
+		// Note that this handling was not possible with the -f segment muxer, since if sometimes doesn't care about the
+		// -segment_times option and outputed segments of 0.2s. With -f segment, we needed to cut at every keyframe.
+		if fpts < (last_frame+OptimalFragmentDuration) {
```

Comment:
This might cause issues with the `-ss` handling of video items.

We might need to have two lists, one for segments and one for keyframes

Reply:
Specially if we have keyframes at 0,5,9.8,10. We might end a segment at 10 & start the next at 9.8